elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -360,9 +357,18 @@ private String extractOrderBy(Class<?> entityClass, Map<Path, Sorting.SortOrder>
 
                     Path.PathElement last = path.lastElement().get();
 
-                    return FilterPredicate.getTypeAlias(last.getType())
+                    String orderByType = FilterPredicate.getTypeAlias(last.getType())
                             + "."
-                            + getColumnName(entityClass, last.getFieldName())
+                            + getColumnName(entityClass, last.getFieldName());
+
+                    if (last.getType() == entityClass) {
+                        Metric currentMetric = schema.getMetric(getColumnName(entityClass, last.getFieldName()));
+                        if (currentMetric != null && query.getMetrics().containsKey(currentMetric)) {
```

Comment:
The else clause for this if statement doesn't really make sense.  What if the column in question is a metric but it isn't in the query metrics - what should we do?  This would be an error and we ought to throw an exception.  However, I don't think it is a good idea to have query validation code scattered everywhere.  We should put that all in one place in a function that validates the query is valid.  All the remaining code can and should assume the query is safe.